backup/backup.go

```diff
@@ -20,6 +21,132 @@ import (
 	"go.uber.org/zap"
 )
 
+type BucketManifestWriter struct {
+	ts *tenant.Service
+	mc *meta.Client
+}
+
+func NewBucketManifestWriter(ts *tenant.Service, mc *meta.Client) BucketManifestWriter {
+	return BucketManifestWriter{
+		ts: ts,
+		mc: mc,
+	}
+}
+
+// WriteManifest writes a bucket manifest describing all of the buckets that exist in the database.
+// It is intended to be used to write to an HTTP response after appropriate measures have been taken
+// to ensure that the request is authorized.
+func (b BucketManifestWriter) WriteManifest(ctx context.Context, w io.Writer) error {
+	bkts, _, err := b.ts.FindBuckets(ctx, influxdb.BucketFilter{})
```

Comment:
Making sure: in a "real" DB this call (and the calls to `OrganizationService`) don't try to take a lock on the KV store, right? i.e. we don't deadlock here

Reply:
They both need read locks, for example the `FindBuckets` goes through [here](https://github.com/influxdata/influxdb/blob/ce536037dc20df4efb686e1093a115f5bc07fd90/tenant/service_bucket.go#L109) & `FindOrganizationByID` go through [here](https://github.com/influxdata/influxdb/blob/ce536037dc20df4efb686e1093a115f5bc07fd90/tenant/service_org.go#L28), both of which end up getting the lock [here](https://github.com/influxdata/influxdb/blob/ce536037dc20df4efb686e1093a115f5bc07fd90/bolt/kv.go#L119).

The `handleBackupMetadata` gets a read lock on the KV at the start of the process and holds it until the end. So once the handler has the lock, the subsequent calls in the process can also get a read lock.

That all sounds great of course, but one other thing I forgot to mention in the initial comment is that I am actually able to successfully hit the `api/v2/backup/metadata` endpoint and everything works fine, so that's some verification that it doesn't deadlock 😄